packages/video_player/video_player/lib/video_player.dart

```diff
@@ -537,6 +549,29 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
     );
   }
 
+  Future<void> _applyMaxVideoResolution() async {
+    if (_isDisposedOrNotInitialized) {
+      return;
+    }
+
+    // Setting the playback speed on iOS will trigger the video to play. We
+    // prevent this from happening by not applying the playback speed until
+    // the video is manually played from Flutter.
+    if (!value.isPlaying) {
+      return;
+    }
+
+    // If we haven't set a max video resolution we won't do anything
```

Comment:
Why is setting a max video resolution ignored if one is not already set? The public method, `setMaxVideoResolution`, should also contain this information.